(dp1
S'output'
p2
S"<type 'exceptions.AttributeError'> 'DAL' object has no attribute 'page'"
p3
sS'layer'
p4
S'/home/abdu/Downloads/web2py/applications/myblog/controllers/default.py'
p5
sS'code'
p6
S"def index():\n    #posts = db(db.post).selectAll()\n    posts = db().select(db.post.id,db.post.title,orderby=db.post.title)\n    return dict(posts=posts)\n\ndef newpost():\n     form = SQLFORM(db.post).process(next=URL('index'))\n     return dict(form=form)\n\n\ndef new():\n     form = SQLFORM(db.post).process(next=URL('index'))\n     return dict(form=form)\n\n@auth.requires_login()\ndef editpost():\n    form = db(db.post,session.args[0])\n    this_post = db.post(request.args(0,cast=int)) or redirect(URL('index'))\n    form = SQLFORM(db.page, this_page).process(next = URL('show',args=request.args))\n    return dict(form=form)\n\ndef user():\n     return dict(form=auth())\n\nresponse._vars=response._caller(new)\n"
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "/home/abdu/Downloads/web2py/gluon/restricted.py", line 217, in restricted\n    exec ccode in environment\n  File "/home/abdu/Downloads/web2py/applications/myblog/controllers/default.py", line 25, in <module>\n  File "/home/abdu/Downloads/web2py/gluon/globals.py", line 372, in <lambda>\n    self._caller = lambda f: f()\n  File "/home/abdu/Downloads/web2py/applications/myblog/controllers/default.py", line 12, in new\n    form = SQLFORM(db.post).process(next=URL(\'index\'))\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 2215, in process\n    self.validate(**kwargs)\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 2154, in validate\n    if self.accepts(**kwargs):\n  File "/home/abdu/Downloads/web2py/gluon/sqlhtml.py", line 1319, in accepts\n    **kwargs\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 2035, in accepts\n    status = self._traverse(status, hideerror)\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 846, in _traverse\n    newstatus = c._traverse(status, hideerror) and newstatus\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 846, in _traverse\n    newstatus = c._traverse(status, hideerror) and newstatus\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 846, in _traverse\n    newstatus = c._traverse(status, hideerror) and newstatus\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 846, in _traverse\n    newstatus = c._traverse(status, hideerror) and newstatus\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 853, in _traverse\n    newstatus = self._validate()\n  File "/home/abdu/Downloads/web2py/gluon/html.py", line 1780, in _validate\n    (value, errors) = validator(value)\n  File "/home/abdu/Downloads/web2py/gluon/validators.py", line 657, in __call__\n    table = self.dbset.db[tablename]\n  File "/home/abdu/Downloads/web2py/gluon/dal.py", line 8034, in __getitem__\n    return self.__getattr__(str(key))\n  File "/home/abdu/Downloads/web2py/gluon/dal.py", line 8041, in __getattr__\n    return ogetattr(self, key)\nAttributeError: \'DAL\' object has no attribute \'page\'\n'
p11
s.